modules/stanford_profile_styles/lib/js/node_form.js

```diff
@@ -0,0 +1,50 @@
+/**
+ * @file
+ * Stops page from changing when user is posting.
+ */
+
+(function ($, Drupal) {
+  'use strict';
+  // Allow Save Submit button.
+  var save = false;
+  // Flag the form has been edited.
+  var edit = false;
+
+  Drupal.behaviors.StanfordProfileStylesNodeProtection = {
+    attach: function (context, settings) {
+      function addListeners() {
+        // Add listeners for all types of data entry.
+        $("input, button, :input", context).once('form-protection').on('blur change click dblclick keydown mousedown select submit', function (e) {
+          edit = true;
+          save = false;
+
+          // Node save button.
+          if ($(this).attr('value') == 'Save') {
+            save = true;
+          }
+        });
+      }
+
+      addListeners();
+      // Periodically add listeners to buttons to check for activity
+      setTimeout(addListeners, 2000);
+
+      // Handle backbutton, exit etc.
+      window.onbeforeunload = function () {
+        // Add CKEditor support.
+        if (typeof (CKEDITOR) != 'undefined' && typeof (CKEDITOR.instances) != 'undefined') {
```

Comment:
Do you have to check the ckeditor itself? I believe it changes the underlying form textarea values so you could just look at all form items. Re: https://git.drupalcode.org/project/confirm_leave/blob/8.x-1.x/js/confirm-leave.js

Reply:
If i remember correctly, when the ckeditor updates the text area, it doesn't trigger listeners. 

I remember there was a reason I put this in and i think it was something like that.

Reply:
This one has a similar approach to yours with the blur on inputs https://git.drupalcode.org/project/js_confirm_pop_up/blob/8.x-1.x/js/js_confirm_pop_up.js

Reply:
one thing to remember is we can't rely on a contrib for our case. because of the react widget, we have to periodically add the listeners when the react form is visible. but we could take the code and modify it